### ISSUE TITLE
SSLCertVerificationError on trying to get_certificate using aio python package

### ISSUE THREAD

**Issue by saisrivathsa** [Azure/azure-sdk-for-python#32649]
- **Package Name**: azure-keyvault
- **Package Version**: 4.20
- **Operating System**: Linux
- **Python Version**: 3.9.14

**Describe the bug**
I am trying to get certificate details using aio python package. I am getting this error on azure kubernetes cluster. The managed identity that I am using for the pod has `Key Vault Certificates Officer` IAM role assignment and also has the required access policies. And I am getting the following error
```
  File "/usr/lib/python3.9/site-packages/azure/core/tracing/decorator_async.py", line 77, in wrapper_use_tracer
    return await func(*args, **kwargs)
  File "/usr/lib/python3.9/site-packages/azure/keyvault/certificates/aio/_client.py", line 154, in get_certificate
    bundle = await self._client.get_certificate(
  File "/usr/lib/python3.9/site-packages/azure/keyvault/certificates/_generated/aio/_operations_mixin.py", line 741, in get_certificate
    return await mixin_instance.get_certificate(vault_base_url, certificate_name, certificate_version, **kwargs)
  File "/usr/lib/python3.9/site-packages/azure/core/tracing/decorator_async.py", line 77, in wrapper_use_tracer
    return await func(*args, **kwargs)
  File "/usr/lib/python3.9/site-packages/azure/keyvault/certificates/_generated/v7_4/aio/operations/_key_vault_client_operations.py", line 1847, in get_certificate
    pipeline_response: PipelineResponse = await self._client._pipeline.run(  # type: ignore # pylint: disable=protected-access
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 221, in run
    return await first_node.send(pipeline_request)
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 69, in send
    response = await self.next.send(request)
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 69, in send
    response = await self.next.send(request)
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 69, in send
    response = await self.next.send(request)
  [Previous line repeated 2 more times]
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/policies/_redirect_async.py", line 73, in send
    response = await self.next.send(request)
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/policies/_retry_async.py", line 205, in send
    raise err
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/policies/_retry_async.py", line 179, in send
    response = await self.next.send(request)
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/policies/_authentication_async.py", line 94, in send
    response = await self.next.send(request)
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 69, in send
    response = await self.next.send(request)
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 69, in send
    response = await self.next.send(request)
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 69, in send
    response = await self.next.send(request)
  [Previous line repeated 2 more times]
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 106, in send
    await self._sender.send(request.http_request, **request.context.options),
  File "/usr/lib/python3.9/site-packages/azure/core/pipeline/transport/_aiohttp.py", line 317, in send
    raise ServiceRequestError(err, error=err) from err
azure.core.exceptions.ServiceRequestError: Cannot connect to host <keyvaultname>.vault.azure.net:443 ssl:True [SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get issuer certificate (_ssl.c:1129)')]

```
**To Reproduce**
1. install azure-identity==1.13.0, azure-keyvault==4.2.0

`
from azure.keyvault.certificates.aio import CertificateClient
from azure.identity.aio import DefaultAzureCredential

azure_credential = DefaultAzureCredential()
certificate_client = CertificateClient(vault_url=<keyvault_url>,
                                           credential=azure_credential)
certificate = await certificate_client.get_certificate(<cerificate_name>)
`

**Expected behavior**
Get certificate should return a KeyVaultCertificate object

 **Screenshots**
![image](https://github.com/Azure/azure-sdk-for-python/assets/15000085/df7d4dc5-04aa-4384-a3d5-1f00074938c8)

**Additional context**
The same code is working fine on local machine and was working fine a couple of days back on the cluster as well but it started giving this error all of a sudden

**Comment by xiangyan99** [comment]
Thanks for the feedback, we’ll investigate asap.

**Comment by xiangyan99** [comment]
@mccoyp the new behavior is possibly related to the change https://github.com/Azure/azure-sdk-for-python/blob/main/sdk/core/azure-core/CHANGELOG.md#1295-2023-10-19

**Comment by mccoyp** [comment]
@saisrivathsa thank you for opening an issue. Do you know what version of `azure-core` is being used in the failing environment?

**Comment by gsrikant7** [comment]
Re-opening this bug as we are facing a similar issue. Please find below the details -

- **Package Name**: azure-keyvault-certificates
- **Package Version**: 4.7.0
- **Operating System**: Linux
- **Python Version**: 3.9.14

We are trying to get certificate details using **azure-keyvault-certificates:4.7.0** azure SDK that in turn uses aiohttp:3.9.2 as async transport. We are getting this error on an azure K8s cluster. The managed identity that we are using for the pod has Key Vault Certificates Officer IAM role assignment and also has the required access policies. We are getting the following error -

```
Request method: 'GET'
Request headers:
    'Accept': 'application/json'
    'x-ms-client-request-id': 'c88c58b4-c5ab-11ee-a5a1-26350f8b0c12'
    'User-Agent': 'azsdk-python-keyvault-certificates/4.7.0 Python/3.9.14 (Linux-5.15.138.1-4.cm2-x86_64-with-glibc2.35)'
No body was attached to the request | 20240205.5 | QueueBasedNotebookExecutor | nb-execution-large
2024-02-07 11:26:53,845 | azure.core.pipeline.policies.http_logging_policy | INFO | Request URL: '[https://dex-jpn-run-kv-ppe2.vault.azure.net/certificates/dex-nbsolution-ppe/?api-version=REDACTED'](https://dex-jpn-run-kv-ppe2.vault.azure.net/certificates/dex-nbsolution-ppe/?api-version=REDACTED%27)
Request method: 'GET'
Request headers:
    'Accept': 'application/json'
    'x-ms-client-request-id': 'c88c58b4-c5ab-11ee-a5a1-26350f8b0c12'
    'User-Agent': 'azsdk-python-keyvault-certificates/4.7.0 Python/3.9.14 (Linux-5.15.138.1-4.cm2-x86_64-with-glibc2.35)'
No body was attached to the request | 20240205.5 | QueueBasedNotebookExecutor | nb-execution-large
Traceback (most recent call last):
  File "/opt/venv/lib/python3.9/site-packages/aiohttp/connector.py", line 992, in _wrap_create_connection
    return await self._loop.create_connection(*args, **kwargs)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 1090, in create_connection
    transport, protocol = await self._create_connection_transport(
  File "/usr/lib/python3.9/asyncio/base_events.py", line 1120, in _create_connection_transport
    await waiter
  File "/usr/lib/python3.9/asyncio/sslproto.py", line 534, in data_received
    ssldata, appdata = self._sslpipe.feed_ssldata(data)
  File "/usr/lib/python3.9/asyncio/sslproto.py", line 188, in feed_ssldata
    self._sslobj.do_handshake()
  File "/usr/lib/python3.9/ssl.py", line 945, in do_handshake
    self._sslobj.do_handshake()
ssl.SSLCertVerificationError: [SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get issuer certificate (_ssl.c:1129)
 
The above exception was the direct cause of the following exception:
 
Traceback (most recent call last):
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/transport/_aiohttp.py", line 263, in send
    result = await self.session.request(  # type: ignore
  File "/opt/venv/lib/python3.9/site-packages/aiohttp/client.py", line 578, in _request
    conn = await self._connector.connect(
  File "/opt/venv/lib/python3.9/site-packages/aiohttp/connector.py", line 544, in connect
    proto = await self._create_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.9/site-packages/aiohttp/connector.py", line 911, in _create_connection
    _, proto = await self._create_direct_connection(req, traces, timeout)
  File "/opt/venv/lib/python3.9/site-packages/aiohttp/connector.py", line 1235, in _create_direct_connection
    raise last_exc
  File "/opt/venv/lib/python3.9/site-packages/aiohttp/connector.py", line 1204, in _create_direct_connection
    transp, proto = await self._wrap_create_connection(
  File "/opt/venv/lib/python3.9/site-packages/aiohttp/connector.py", line 994, in _wrap_create_connection
    raise ClientConnectorCertificateError(req.connection_key, exc) from exc
aiohttp.client_exceptions.ClientConnectorCertificateError: Cannot connect to host dex-jpn-run-kv-ppe2.vault.azure.net:443 ssl:True [SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get issuer certificate (_ssl.c:1129)')]
 
The above exception was the direct cause of the following exception:
 
Traceback (most recent call last):
  File "/src/app/notebook_executor/notebook_queue_based_executor/./main.py", line 258, in <module>
    asyncio.run(main())
  File "/usr/lib/python3.9/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/usr/lib/python3.9/asyncio/base_events.py", line 647, in run_until_complete
    return future.result()
  File "/src/app/notebook_executor/notebook_queue_based_executor/./main.py", line 90, in main
    confidential_client_application = await create_aad_confidential_client_application(azure_credential,
  File "/src/app/notebook_executor/notebook_queue_based_executor/./main.py", line 176, in create_aad_confidential_client_application
    certificate = await certificate_client.get_certificate(aad_app_configuration.certificate_name)
  File "/opt/venv/lib/python3.9/site-packages/azure/core/tracing/decorator_async.py", line 77, in wrapper_use_tracer
    return await func(*args, **kwargs)
  File "/opt/venv/lib/python3.9/site-packages/azure/keyvault/certificates/aio/_client.py", line 154, in get_certificate
    bundle = await self._client.get_certificate(
  File "/opt/venv/lib/python3.9/site-packages/azure/keyvault/certificates/_generated/aio/_operations_mixin.py", line 741, in get_certificate
    return await mixin_instance.get_certificate(vault_base_url, certificate_name, certificate_version, **kwargs)
  File "/opt/venv/lib/python3.9/site-packages/azure/core/tracing/decorator_async.py", line 77, in wrapper_use_tracer
    return await func(*args, **kwargs)
  File "/opt/venv/lib/python3.9/site-packages/azure/keyvault/certificates/_generated/v7_4/aio/operations/_key_vault_client_operations.py", line 1847, in get_certificate
    pipeline_response: PipelineResponse = await self._client._pipeline.run(  # type: ignore # pylint: disable=protected-access
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 221, in run
    return await first_node.send(pipeline_request)
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 69, in send
    response = await self.next.send(request)
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 69, in send
    response = await self.next.send(request)
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 69, in send
    response = await self.next.send(request)
  [Previous line repeated 2 more times]
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/policies/_redirect_async.py", line 73, in send
    response = await self.next.send(request)
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/policies/_retry_async.py", line 205, in send
    raise err
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/policies/_retry_async.py", line 179, in send
    response = await self.next.send(request)
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/policies/_authentication_async.py", line 94, in send
    response = await self.next.send(request)
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 69, in send
    response = await self.next.send(request)
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 69, in send
    response = await self.next.send(request)
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 69, in send
    response = await self.next.send(request)
  [Previous line repeated 2 more times]
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/_base_async.py", line 106, in send
    await self._sender.send(request.http_request, **request.context.options),
  File "/opt/venv/lib/python3.9/site-packages/azure/core/pipeline/transport/_aiohttp.py", line 300, in send
    raise ServiceRequestError(err, error=err) from err
azure.core.exceptions.ServiceRequestError: Cannot connect to host dex-jpn-run-kv-ppe2.vault.azure.net:443 ssl:True [SSLCertVerificationError: (1, '[SSL: CERTIFICATE_VERIFY_FAILED] certificate verify failed: unable to get issuer certificate (_ssl.c:1129)')]
Stream closed EOF for jpn/notebook-executor-largesku-8689ffbdb-przg5 (wdatp-service)
```

**To Reproduce**
Steps to reproduce the behavior - 

Install **azure-identity==1.13.0**, **azure-keyvault==4.2.0**, **azure-keyvault-certificates==4.7.0**, and **aiohttp==3.9.2 / 3.9.3**

```
from azure.identity.aio import DefaultAzureCredential
from azure.keyvault.certificates.aio import CertificateClient

azure_credential = DefaultAzureCredential()
certificate_client = CertificateClient(vault_url=<keyvault_url>, credential=azure_credential)
certificate = await certificate_client.get_certificate(<cerificate_name>)
```

**Expected behavior**
**get_certificate** should return a KeyVaultCertificate object. Instead, we get the above error.

**Additional Context**

- The same code is working fine on a local machine (windows).
- We moved from using **aiohttp==3.9.0** to **aiohttp==3.9.2** and we see that this [change](https://github.com/aio-libs/aiohttp/pull/7698) in 3.9.2, fiddles with the SSL parameter.
